core/src/main/java/org/mule/transport/AbstractTransportMessageProcessTemplate.java

```diff
@@ -43,8 +43,20 @@ public AbstractTransportMessageProcessTemplate(MessageReceiverType messageReceiv
 
     public MuleEvent getMuleEvent() throws MuleException
     {
-        MuleMessage messageFromSource = createMessageFromSource(getOriginalMessage());
-        return createEventFromMuleMessage(messageFromSource);
+        if (muleEvent == null)
+        {
+            MuleMessage messageFromSource = createMessageFromSource(getOriginalMessage());
+            muleEvent = createEventFromMuleMessage(messageFromSource);
+        }
+        else
+        {
+            MuleMessage message = muleEvent.getMessage();
+            if(message instanceof ThreadSafeAccess)
+            {
+                ((ThreadSafeAccess)message).resetAccessControl();
```

Comment:
Since getMuleEvent() may be called by different threads. For instance, Throttling may get the event, do some processing, and then retry again later in a different thread.